### DISCUSSION TITLE
Emails falling into spam

### DISCUSSION THREAD

**Discussion by WendelLuanDev** [pocketbase/pocketbase#5243] [Q&A]
My email confirmation and password change emails always go to the spam folder, I want to know if there is any configuration I can do, so that this doesn't happen, if this is not possible, I can trigger my own verification emails with its own node service? I would need to capture the email acceptance link that pocketbase itself has and send it to my email, I saw that there is a hook that runs before sending a verification email, which is this: but it is possible to get this link or a token to send these confirmation emails?

**Comment by ganigeorgiev** [comment]
The spam score could be affected by a lot of things, but most likely it would depend on your mail server and its configuration - IP reputation, DMARC, DKIM, SPF DNS records, etc.

Based on my tests locally with the default Spamassasin filter the PocketBase emails spam score is 0. This could be also confirmed with the mailtrap.io "Spam analysis" check:

![spam_analysis](https://i.imgur.com/fyr5G30.png)

> I can trigger my own verification emails with its own node service?

You can but I don't think it will make any difference.
If you are using the JSVM pb_hooks you can check the [`$tokens.*`](https://pocketbase.io/jsvm/modules/_tokens.html) and [`$mails.*`](https://pocketbase.io/jsvm/modules/_mails.html) helpers.
For Go the same above helpers are under the [`tokens`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.17/tokens) and [`mails`](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.17/mails) subpackages.

My recommendation is to check your DNS email related records and/or contact your mail service provider to see if there is anything they can assist you with (if you are using a shared mail server then it is also possible your mail server IP to be blacklisted).

**Reply by ganigeorgiev** [reply]
To verify that your mail server is properly configured you can also send a test mail from the _Admin UI > Settings > Mail settings > "Send test mail"_ for example to the generated address from https://www.mail-tester.com/ or other similar websites and review the analysis.

**Comment by WendelLuanDev** [comment]
I'm using sendingblue services, can you recommend a good and consistent email server? Even if paid

**Reply by ganigeorgiev** [reply]
I've used Sendinblue/Brevo in the past and it was fine.
Make sure that you have properly setup the DNS verification records for your domain (as mentioned earlier in https://github.com/pocketbase/pocketbase/discussions/5243#discussioncomment-10113763 you can confirm that it works correctly by sending a test email from the Admin UI to https://www.mail-tester.com).

Other free mail services that I've tried in the past:
- https://www.mailjet.com/
- https://sendgrid.com/
- https://www.mailgun.com/
- https://mailtrap.io/email-sending/ (_this is a new offering and I haven't tested it in production yet but I often use their Email Testing service during development_)

**Comment by WendelLuanDev** [comment]
![image](https://github.com/user-attachments/assets/75083ea2-8a3e-478f-aeeb-561439554255)
 do i got this, on https://www.mail-tester.com/ this configs, can i do this on pocketbase? (DMARC configs)

**Reply by ganigeorgiev** [reply]
This is not really PocketBase related as we don't have control over your DNS mail records.

You can check the Brevo instructional video how to setup the DMARC record policy - https://help.brevo.com/hc/en-us/articles/12163873383186-Authenticate-your-domain-with-Brevo-Brevo-code-DKIM-record-DMARC-record.
